bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/findandreplace/IFindReplaceLogic.java

```diff
@@ -0,0 +1,230 @@
+package org.eclipse.ui.findandreplace;
+
+import org.eclipse.swt.widgets.Display;
+
+import org.eclipse.jface.text.IFindReplaceTarget;
+
+public interface IFindReplaceLogic {
+
+	/**
+	 * A global search in files searches in the whole file, as opposed to only
+	 * searching in a part of the selection.
+	 *
+	 * @return boolean whether the search is global
+	 */
+	boolean isGlobalSearch();
```

Comment:
Might it make sense to have an enum with all the configuration options and only methods `activate(option)`, `deactivate(option)` and `isActive(option)`? The interface would become less verbose then.

Reply:
There are currently only two states:
- Global search
- Local search (only search in the selected scope)

It is possible that in the future we allow for more search-scopes.
Imagine searching in the outline. You could allow for
- Global Search
- Selection Search (search in selected)
- Method search (search in methods)
- Field search (search in fields)
- Type search (search in types)

In the first case, I would argue that the enum might even obfuscate the code more than it would help understand it. However, in the second case, the enum would be a good solution. What do you think?

Reply:
My comment was obviously a little imprecise: I did not propose to provide one enum per option but one enum with all two-valued options, i.e., something like:
```java
enum SearchOption {
  GLOBAL,
  WRAP,
  CASE_SENSITIVE,
  WHOLE_WORD
}
```
and then have methods `activate(SearchOption)`, `deactivate(SearchOption)` and `isActive(SearchOption`). In case you add an option, the only API change will be an extension of the enum instead of adding three methods for each option.